docs/reference/glossary.asciidoc

```diff
@@ -105,6 +115,20 @@
   +
   See also <<glossary-routing,routing>>
 
+[[glossary-query]] query ::
+
+  A query is the basic component of a search. A search can be defined by one or more queries 
+  which can be mixed and matched in endless combinations. The term Query refers to all queries 
+  which not only determine if a document matches, but also calculate how well the document matches. 
+  This calculation is refered to as scoring, hence these queries are also known as "scoring queries". 
+  A scoring query calculates how relevant each document is to the query, and assigns 
+  it a relevance score, which is later used to sort matching documents by relevance. 
+  This concept of relevance is well suited to full-text search, where there is seldom a 
+  completely “correct” answer. These queries are takes more resources than 
+  <<glossary-filter,non scoring queries>> and their query results are not cacheable. 
+  As a general rule, use query clauses for full-text search or for any condition that should 
+  affect the relevance score, and use filters for everything else.
```

Comment:
maybe user "condition that requires scoring" if you agree on leaving you the notion of relevance here.